### PR TITLE
fix: moved mutual functions to utils, updated trims filter

### PR DESCRIPTION
**Pull request by minhuyyyy**
image

### CODE REVIEW

File: src/components/modals/desktop-filter-dropdown.tsx [hunk 1]
```diff
@@ -1,70 +1,59 @@
...
-    };
-
-    const handleYearChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-        e.currentTarget.id === 'min-year'
-            ? setYear([(year[0] = parseInt(e.currentTarget.value)), year[1]])
-            : setYear([year[0], (year[1] = parseInt(e.currentTarget.value))]);
+    const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+        const { id, value } = e.currentTarget;
+        const splittedId = id.split('-');
+        splittedId[0] === 'min'
```
Comment (ttnguyen115) [inline src/components/modals/desktop-filter-dropdown.tsx:46]: should use if else for more readable

File: src/components/modals/desktop-filter-dropdown.tsx [hunk 2]
```diff
@@ -96,7 +85,7 @@ function DesktopFilterDropdown({ components }) {
     const getMakeName = (makeId: number) => {
         setSelectedCar({ ...selectedCar, make_id: makeId });
         const make = makes.find((make: Make) => make.id === makeId);
-        setSelectedCar({ ...selectedCar, make_name: make?.name, name: '' });
+        setSelectedCar({ ...selectedCar, make_name: make!.name, name: '' });
```
Comment (ttnguyen115) [inline src/components/modals/desktop-filter-dropdown.tsx:88]: non-null assertion ! - can be dangerous if cannot get `makes` from api request or redux, consider refactor this one

File: src/components/modals/desktop-filter-dropdown.tsx [hunk 3]
```diff
@@ -111,17 +100,12 @@ function DesktopFilterDropdown({ components }) {
         const res = await trimsFetcher(
             `/trims?verbose=yes&year=2019&make_model_id=${modelId}&make_id=${makeId}`,
         );
-        console.log(res.data);
     };
 
     useEffect(() => {
         getMakeCars();
```
Comment (ttnguyen115) [inline src/components/modals/desktop-filter-dropdown.tsx:106]: find way to bring it to SSR

File: src/components/modals/mobile-filter-modal.tsx [hunk 4]
```diff
@@ -21,15 +21,17 @@ function MobileFilterModal() {
                     dispatch(
```
Comment (ttnguyen115) [inline src/components/modals/mobile-filter-modal.tsx:21]: `dispatch` should not be defined in anonymous function

File: src/components/modals/mobile-filter-modal.tsx [hunk 5]
```diff
@@ -21,15 +21,17 @@ function MobileFilterModal() {
...
                         duckCreator.creators.setShowMobileFilterModal(true),
                     )
-                }>
+                }
+            >
                 Filter
             </div>
             <div
                 className={`w-full h-full bg-white fixed top-0 left-0 transition-transform duration-300 ${
```
Comment (ttnguyen115) [inline src/components/modals/mobile-filter-modal.tsx:29]: can install and use lib `clsx` for className handler
https://www.npmjs.com/package/clsx

File: src/components/shared/navigations/navbar.tsx [hunk 6]
```diff
@@ -1,27 +1,24 @@
...
 function NavbarComponent() {
-    const [isMenuOpen, setIsMenuOpen] = React.useState(false);
+    const [isMenuOpen, setIsMenuOpen] = useState(false);
 
     const ref = useRef(null);
 
     useEffect(() => {
         const handleOutSideClick = (event) => {
-            if (!ref.current?.contains(event.target)) {
+            if (ref.current != event.target) {
```
Comment (ttnguyen115) [inline src/components/shared/navigations/navbar.tsx:21]: consider to use !== if applicable

File: src/redux/selectors.ts
Comment (ttnguyen115) [inline src/redux/selectors.ts:1]: split selectors into smaller selector files in order to easily manage when scaling app later

Not: makes + cars + filters + showMobile + ...

**Review by ttnguyen115 — COMMENTED** [review]
Overall, should place name file as camelCase

**Comment by ttnguyen115** [comment]
should add UI snapshots for PR description

**Comment by minhuyyyy** [comment]
image

//updated trims filter

File: src/components/modals/DesktopFilterDropdown.tsx [hunk 7]
```diff
@@ -132,17 +124,21 @@ function DesktopFilterDropdown({ components }) {
...
                                 placeholder='Min'
                                 value={year[0] || 0}
                                 className='flex-col-reverse'
                             />
                             <TextInput
+                                inputStyle='w-4/5'
                                 filterName='Max'
                                 id='year'
-                                onChange={handleYearChange}
+                                onChange={(e) => handleInputChange(e)}
```
Comment (ttnguyen115) [inline src/components/modals/DesktopFilterDropdown.tsx:141]: if passing only event parameter, should use `onChange={handleInputChange}` only

File: src/components/modals/DesktopFilterDropdown.tsx [hunk 8]
```diff
@@ -291,16 +288,19 @@ function DesktopFilterDropdown({ components }) {
...
-                                        filterName={item}
-                                        id={`body-${index + 1}`}
-                                        onChange={() => {}}
-                                        // className='flex-row items-start'
-                                    />
-                                </div>
-                            ))}
+                            {filters.bodiestype.map(
+                                (item: string, index: number) => (
+                                    <div key={index}>
```
Comment (ttnguyen115) [inline src/components/modals/DesktopFilterDropdown.tsx:293]: consider to use <Fragment> with key if `div` has no special properties + UI is not affected

File: src/components/shared/carHolder/carHolder.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+//components
+import CarCard from '../carCard/carCard';
+
+//types
+import { Car } from '@contracts/types/car';
+
+//redux-selectors
+import { getCars, getMakes } from '@redux/selectors';
+CarCard;
```
Comment (ttnguyen115) [inline src/components/shared/carHolder/carHolder.tsx:18]: review and clean code carefully

File: src/components/shared/carHolder/carHolder.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+function CarHolder() {
+    let cars = getCars();
```
Comment (ttnguyen115) [inline src/components/shared/carHolder/carHolder.tsx:20]: consider to change `let` to `const`

File: src/components/shared/filters/checkboxFilter.tsx [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+import MobileFilterModal from '@components/modals/MobileFilterModal';
+
+//redux-selectors
+import { getShowMobile } from '@redux/selectors';
+
+function CheckboxFilter() {
+    const showMobile = getShowMobile();
+
+    return (
+        <div>
```
Comment (ttnguyen115) [inline src/components/shared/filters/checkboxFilter.tsx:14]: consider to use Fragment

File: src/pages/_app.jsx [hunk 12]
```diff
@@ -1,23 +1,23 @@
...
+import duckCreator from '@ducks/duckCreator';
 
 // redux
+import { getDuckEgg, wrapperInitializer } from '@redux/store';
 
-// fetch
+// fetchers
+import filtersFetcher from '@fetchers/filterFetcher';
+import { seedingData } from '../utils';
+import { carsFetcher } from '@fetchers/index';
```
Comment (ttnguyen115) [inline src/pages/_app.jsx:11]: if carsFetcher in index file, remove /index

**Review by ttnguyen115 — COMMENTED** [review]
review 11/07/2024

**Review by ttnguyen115 — APPROVED** [review]
looks good to me

**Pull request merged**